crates/noirc_abi/src/input_parser/toml.rs

```diff
@@ -55,9 +55,14 @@ fn toml_map_to_field(
     for (parameter, value) in toml_map {
         let mapped_value = match value {
             TomlTypes::String(string) => {
-                let new_value = parse_str(&string)?;
-
-                if let Some(field_element) = new_value {
+                let new_value = try_str_to_field(&string);
+
+                // We accept UTF-8 strings as input as well as hex strings representing field elements.
+                // We still want developers to be able to pass in hex strings for FieldElement inputs.
+                // Thus, we first try to parse the string into a field element, and if that fails we assume that the input is meant to be a plain string
```

Comment:
Is this what we want for strings though? Wouldn't this prevent a user from passing in the string `"0x41"` by translating it to `"A"` instead?

Reply:
Not entirely I guess as users cannot pass any utf8 string as input to a circuit. It would prevent a user from passing in the string "0x41" because it will be converted to a field element by the input parser, rather than translating it to "A". They can technically still have "0x41" as input but it must be a Field.

I mainly went with this method because of this comment `We still want developers to be able to pass in hex strings for FieldElement inputs.` Otherwise, we would have to add some more details to our input parser. We do compile our circuit before reading the inputs from file. I could add some logic where I pass along the ABI and then parse accordingly.

Reply:
We'd still like that ability, but I think this should be type directed so that we can have both. It would be very confusing for users if they pass in something within quotes to a string field and the value of that string was not exactly what they wrote within the quotes.

Reply:
Fair point. I will look into passing the Abi when we parse the toml so that we can differentiate between parsing into a field element or a string

Reply:
I have now made it now so that the parsing for `TomlType::String` is directed by the abi types. I also added some extra inputs to my `strings` test using a string input and a field input for `0x41`.